packages/node_modules/@ciscospark/spark-core/test/unit/spec/interceptors/rate-limit.js

```diff
@@ -0,0 +1,291 @@
+/*!
+ * Copyright (c) 2015-2017 Cisco Systems, Inc. See LICENSE file.
+ */
+
+/* eslint-disable camelcase */
+import {assert} from '@ciscospark/test-helper-chai';
+import MockSpark from '@ciscospark/test-helper-mock-spark';
+import {
+  RateLimitInterceptor,
+  config,
+  Credentials,
+  SparkHttpError,
+  Token
+} from '@ciscospark/spark-core';
+import {cloneDeep} from 'lodash';
+describe(`spark-core`, () => {
+  describe(`Interceptors`, () => {
+    describe(`RateLimitInterceptor`, () => {
+
+      function sleep(time) {
+        return new Promise((resolve) => setTimeout(resolve, time));
+      }
+
+      let interceptor, spark;
+      beforeEach(() => {
+        spark = new MockSpark({
+          children: {
+            credentials: Credentials
+          },
+          config: cloneDeep(config)
+        });
+
+        spark.credentials.supertoken = new Token({
+          access_token: `ST1`,
+          token_type: `Bearer`
+        }, {parent: spark});
+
+        interceptor = Reflect.apply(RateLimitInterceptor.create, spark, []);
+      });
+
+      describe(`#isRateLimited`, () => {
+        it(`returns false when URI is not IDbroker URI`, () => {
+          assert.equal(interceptor.isRateLimited(`https://fakeuir.com/testFake`), false);
+        });
+        it(`returns false if the URI is null`, () => {
+          assert.equal(interceptor.isRateLimited(null), false);
+        });
+        it(`returns true when idbroker API is rate limited`, () => {
+          interceptor.setRateLimitExpiry(`https://idbroker.webex.com/horse/v1/myID`, new Date().getTime() * 2);
+          assert.equal(interceptor.isRateLimited(`https://idbroker.webex.com/horse/v1/myID`), true);
+        });
+        it(`returns true when identity API is rate limited`, () => {
+          interceptor.setRateLimitExpiry(`https://identity.webex.com/horse/v1/myID`, new Date().getTime() * 2);
+          assert.equal(interceptor.isRateLimited(`https://identity.webex.com/horse/v1/myID`), true);
+        });
+        it(`returns false when previously rate limited idbroker API expiry time has been met`, () => {
+          interceptor.setRateLimitExpiry(`https://idbroker.webex.com/horse/v1/myID`, 1);
+          return sleep(2000)
```

Comment:
Take a look at our use of `lolyx` for mock timers so that tests can be written synchronously (and run significantly faster).